frontend/packages/kubevirt-plugin/src/utils/sort.ts

```diff
@@ -46,3 +47,74 @@ export const ignoreCaseSort = <T>(
   };
   return array.sort((a, b) => resolve(a).localeCompare(resolve(b)));
 };
+
+const getOSVersion = (osName: string): number[] =>
+  osName
+    .split(/\D/)
+    .filter((x) => x)
+    .map((num) => parseInt(num, 10));
+
+/**
+ *
+ *
+ * Compare the numbers between the two versions by the order of their appearance
+ * in the OS name.
+ *
+ * For example:
+ * osVersion1: [10,2] for OS: 'Windows 10 R2',
+ * osVersion2: [10] for OS: 'Windows 10',
+ * (return 1)
+ *
+ * osVersion1: [9,10] for OS: 'ubuntu9.10',
+ * osVersion2: [10,4] for OS: 'ubuntu10.04',
+ * (return -1)
+ *
+ * return 0 when equal.
+ *
+ */
+export const compareVersions = (osVersion1: number[], osVersion2: number[]): number => {
+  if (!osVersion1 && !osVersion2) {
+    return 0;
+  }
+
+  const osVer1 = osVersion1 || [0];
+  const osVer2 = osVersion2 || [0];
+
+  const zipped = _.zip(osVer1, osVer2);
+  let idx = 0;
+  while (idx < zipped.length) {
+    /*
+      undefined values are equal to 0, eg:
+      14.0 == 14 -> zipped = [[14,14],[0,undefined]]
+      1.0.0 == 1 -> zipped = [[1,1],[0,undefined],[0,undefined]]
+    */
+    const ver1 = !zipped[idx][0] ? 0 : zipped[idx][0];
+    const ver2 = !zipped[idx][1] ? 0 : zipped[idx][1];
+
+    if (ver1 > ver2) {
+      return 1;
+    }
+
+    if (ver2 > ver1) {
+      return -1;
+    }
+
+    idx++;
+  }
+
+  return 0;
+};
+
+const descSortOSes = (os1: OperatingSystemRecord, os2: OperatingSystemRecord): number => {
+  if (os1.name.localeCompare(os2.name) > 0) {
+    return -1;
+  }
+  if (os1.name.localeCompare(os2.name) < 0) {
+    return 1;
+  }
+
+  return compareVersions(getOSVersion(os1.id), getOSVersion(os2.id)) * -1;
+};
+
+export const removeOSDups = (osArr: OperatingSystemRecord[]): OperatingSystemRecord[] =>
+  _.uniqBy([...osArr].sort(descSortOSes), 'name');
```

Comment:
```suggestion
  _.uniqBy(osArr.filter(a => a).sort(descSortOSes), 'name');
```

I guess this makes it more useful and we get a new array as well